__integrations__/__tests__/cosmoscdcservice.test.ts

```diff
@@ -0,0 +1,340 @@
+    TE.getOrElse((e) => {
+      throw Error(
+        `Cannot initialize integration tests - ${JSON.stringify(e.message)}`,
+      );
+    }),
+  )();
+}, 60000);
+
+afterAll(async () => {
+  await pipe(
+    deleteDatabase(cosmosClient, COSMOSDB_NAME),
+    TE.getOrElse((e) => {
+      throw Error(`Cannot delete db ${e.message}`);
+    }),
+  )();
+});
+
+afterEach(async () => {
+  setShouldExitExternal(false);
+});
+
+const processResults = (
+  _: ReadonlyArray<unknown>,
+): TE.TaskEither<Error, void> => {
+  return TE.right(void 0);
+};
+
+describe("cosmosCDCService", () => {
+  it("should handle incorrect ConnectionString - Error", async () => {
+    const result = await service.processChangeFeed(
+      new CosmosClient(
+        `AccountEndpoint=https://localhost:8081;AccountKey=myAccountKey123;`,
+      ),
+      COSMOSDB_NAME,
+      COSMOS_COLLECTION_NAME,
+      processResults,
+    )(cosmosDBService)();
+    expect(E.isLeft(result)).toBeTruthy();
+  });
+
+  it("should handle incorrect database name - Error", async () => {
+    const result = await pipe(
+      cosmosDBService.connect({
+        connection: COSMOSDB_CONNECTION_STRING,
+      }),
+      TE.chain((client) =>
+        service.processChangeFeed(
+          client,
+          "not-existing-database",
+          COSMOS_COLLECTION_NAME,
+          processResults,
+        )(cosmosDBService),
+      ),
+    )();
+    expect(result).toEqual(
+      E.left(new Error(`Impossible to get database not-existing-database`)),
+    );
+  });
+
+  it("should handle incorrect container name - Error", async () => {
+    const result = await pipe(
+      cosmosDBService.connect({
+        connection: COSMOSDB_CONNECTION_STRING,
+      }),
+      TE.chain((client) =>
+        service.processChangeFeed(
+          client,
+          COSMOSDB_NAME,
+          "not-existing-collection",
+          processResults,
+        )(cosmosDBService),
+      ),
+    )();
+    expect(result).toEqual(
+      E.left(new Error(`Impossible to get container not-existing-collection`)),
+    );
+  });
+
+  it("should handle incorrect lease container name - Error", async () => {
+    const result = await pipe(
+      cosmosDBService.connect({
+        connection: COSMOSDB_CONNECTION_STRING,
+      }),
+      TE.chain((client) =>
+        service.processChangeFeed(
+          client,
+          COSMOSDB_NAME,
+          COSMOS_COLLECTION_NAME,
+          processResults,
+          "not-existing-lease-container",
+        )(cosmosDBService),
+      ),
+    )();
+    expect(result).toEqual(
+      E.left(
+        new Error(`Impossible to get container not-existing-lease-container`),
+      ),
+    );
+  });
+
+  it("should process table content starting from beginning - no continuation token, default lease container", async () => {
+    //Forcing change feed processor to exit after 8 seconds
+    setTimeout(() => {
+      setShouldExitExternal(true);
+    }, 8000);
```

Comment:
maybe we can wrap promise timeout instead of set an instance's boolean variable. Something like this:
```
const promiseWithTimeout = <T>(
  promise: Promise<T>,
  ms: number,
  timeoutError = new Error('Promise timed out')
) {
  // create a promise that rejects in milliseconds
  const timeout = new Promise((_, reject) => {
    setTimeout(() => {
      reject(timeoutError);
    }, ms);
  });

  // returns a race between timeout and the passed promise
  return Promise.race([promise, timeout]);
}
```
then we can apply timeout on a pipe execution i.e:
```
await promiseWithTimeout(
   pipe(
      cosmosDBService.connect({
        connection: COSMOSDB_CONNECTION_STRING,
      }),
      TE.chain((cosmosClient) =>
        service.processChangeFeed(
          cosmosClient,
          COSMOSDB_NAME,
          COSMOS_COLLECTION_NAME,
          processResults,
        )(cosmosDBService),
      ),
    )(), 
   8000
);
```